### DISCUSSION TITLE
esp32 c3-s5 dm9051

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#16132] [ESP32]
Hy.
I am a newbie into esp32 and micropython.
I tried to use my ethernet port poe.

By collecting informations and helped by copilot i got this code : 

```
import network
import machine
import time

# Configuration des broches SPI pour le DM9051
CS_PIN = 9
MISO_PIN = 3
MOSI_PIN = 10
SCK_PIN = 7

# Initialisation de la communication SPI
spi = machine.SPI(1, baudrate=10000000, polarity=0, phase=0, sck=machine.Pin(SCK_PIN), mosi=machine.Pin(MOSI_PIN), miso=machine.Pin(MISO_PIN))

# Initialisation du module Ethernet
try:
    eth = network.LAN(spi=spi, cs=machine.Pin(CS_PIN), phy_type=network.PHY_DM9051, phy_addr=1)
except Exception as e:
    print("Failed to initialize Ethernet:", e)
    raise

# Configuration de l'adresse MAC (doit être unique sur le réseau)
try:
    eth.config(mac=b'\xDE\xAD\xBE\xEF\xFE\xED')
except Exception as e:
    print("Failed to configure Ethernet:", e)
    raise

# Activation de l'interface Ethernet
try:
    eth.active(True)
except Exception as e:
    print("Failed to activate Ethernet:", e)
    raise

```

But i got this error : 

```
Failed to initialize Ethernet: esp_eth_driver_install failed with invalid argument
Traceback (most recent call last):
  File "boot.py", line 16, in <module>
OSError: esp_eth_driver_install failed with invalid argument
```

I don't manage to see where i am wrong.
Can somebody help me, please :)?

Thanks by Advacne

**Comment by Josverl** [comment]
Can you provide details on the exact port, nic and Micropython firmware you are running this on?

**Reply by ghost** [reply]
Hy.
First, thank you to asnwer me.

-The firmware version : i tried with 1.24 and the preveiw 1.25.

- The esp32 i bought is this one :esp32 eth01-evo (wt32c3-s5)
https://fr.aliexpress.com/item/1005006895984276.html?spm=a2g0o.order_list.order_list_main.15.2b0b5e5bdmUQXb&gatewayAdapt=glo2fra

-For the ports : 
SPI_CN : 9
SPI_CLK : 7
SPI_MOSI : 10
 SPI_MISO : 3
INT : 8
RSTB : 6

spi = machine.SPI(1, baudrate=10000000, polarity=0, phase=0, sck=machine.Pin(SCK_PIN), mosi=machine.Pin(MOSI_PIN), miso=machine.Pin(MISO_PIN))

eth = network.LAN(spi=spi, cs=machine.Pin(CS_PIN), phy_type=network.PHY_DM9051, phy_addr=1)

**Reply by scruss** [reply]
Do the following commands give an error?
```python3
import network
print(network.PHY_DM9051)
```
If so, the firmware you are using may not have DM9051 support compiled in. v1.24.0 for [ESP32_GENERIC_C3](https://micropython.org/download/ESP32_GENERIC_C3/)  does seem to have this, though.

**Reply by ghost** [reply]
Hy ^^
Thanks you for your Answer.
I have dm9051 in the firmware. 
I tried your command and i Can see it with command  dir(network), After importing network.

**Comment by bidrohini** [comment]
Confirm that your Ethernet PHY module indeed uses the DM9051. If it uses a different chip, you'll need to set the correct PHY type in your code.

**Reply by ghost** [reply]
Hy.
I confirm for dm9051.
